src/components/molecules/TodoItem.vue

```diff
@@ -0,0 +1,34 @@
+<script setup>
+import AppCheckboxInput from '@/components/atoms/AppCheckboxInput.vue'
+import AppButton from '@/components/atoms/AppButton.vue'
+import { ref } from 'vue'
+
+const props = defineProps({
+  name: { type: String, required: true },
+  completed: { type: Boolean, default: false },
+})
+
+const isHovering = ref(false)
+
+const textDecoration = (isCompleted) => (isCompleted ? { textDecoration: 'line-through', color: '#808080' } : {})
+</script>
+
+<template>
+  <div :class="$style['todo-item']" @mouseover="isHovering = true" @mouseout="isHovering = false">
+    <AppCheckboxInput :checked="props.completed" />
+    <div>
+      <span :style="textDecoration(props.completed)">{{ props.name }}</span>
+      <AppButton v-show="isHovering" label="X" />
```

Comment:
I think you can use the :hover selector here too. And set display: none and display: block for this button when hovering and removing the cursor